core/src/main/kotlin/org/evomaster/core/problem/externalservice/httpws/service/HarvestActualHttpWsResponseHandler.kt

```diff
@@ -427,4 +446,64 @@ class HarvestActualHttpWsResponseHandler {
             return false
         }
     }
+
+    /**
+     * Finds the closest harvested requested based on the given key.
+     * Uses Levenshtein Distance to calculate the distance, then selects the
+     * shortest. If none exists, returns null.
+     */
+    private fun findClosestRequest(key: String): String? {
+        var out: String? = null
+        var diff = 100.0
+
+        actualResponses.forEach { (k, v) ->
+            val httpWsResponseParam = v.param as HttpWsResponseParam
+
+            if (matchRequest(key, k)) {
+                if (httpWsResponseParam.status.values[httpWsResponseParam.status.index] == 200) {
+                    val ld = LevenshteinDistance.getDefaultInstance().apply(k, key).toDouble()
+
+                    if (ld == 0.0) {
+                        out = k
+                    } else {
+                        val nDiff = (ld / key.length.toDouble()) * 100.0
+
+                        if (nDiff < diff) {
+                            diff = nDiff
+                            out = k
+                        }
+                    }
+                }
+            }
+        }
+        return out
+    }
+
+    private fun getURL(url: String): List<String> {
+        return url.split("::")
+    }
+
+    private fun getMethod(url: String): String {
+        return url.split("::")[0].lowercase()
+    }
```

Comment:
these functions are making major assumptions on the structure of the URL... replace with JDK API `URL` objects, which is safer and more complete